nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
I started down the path of also adding this logic to `nvm install` but to your point, I'm thinking I should back that out and only add this logic to `nvm use`. If you agree with that, then do you still think this would need to be implemented without node?